packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DocumentController.java

```diff
@@ -128,12 +128,12 @@ public ResponseEntity<List<DocumentAsset>> getDocuments(
 			@ApiResponse(responseCode = "200", description = "Document created.", content = @Content(mediaType = "application/json", schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = ResponseId.class))),
 			@ApiResponse(responseCode = "500", description = "There was an issue creating the document", content = @Content)
 	})
-	public ResponseEntity<ResponseId> createDocument(
+	public ResponseEntity<DocumentAsset> createDocument(
 			@RequestBody final DocumentAsset document) {
 
 		try {
-			documentAssetService.createDocumentAsset(document);
-			return ResponseEntity.ok(new ResponseId(document.getId()));
+			DocumentAsset d = documentAssetService.createDocumentAsset(document);
```

Comment:
how come this is not following the same pattern as the other controllers, where you removed the final from the parameter and update it?

Reply:
It was the first one I came across with `final` - i should change that